### DISCUSSION TITLE
Changing dns provider for acme client

### DISCUSSION THREAD

**Discussion by AnFerrr** [0xJacky/nginx-ui#113] [Q&A] [answered]
Hey! 

I'm currently trying to get a wildcard certificate from let's encrypt through nginx-ui, using a DNS challenge.

I get an error upon asking the certificate. Here is the output:

`Getting the certificate, please wait...

Generating private key for registering account

Preparing lego configurations

Creating client facilitates communication with the CA server

Using DNS01 challenge provider

Setting environment variables

Registering user

Obtaining certificate

Cleaning environment variables

fail to obtain: error: one or more domains had a problem: [*.drive.domain.fr] [*.drive.domain.fr] acme: could not determine solvers [drive.domain.fr] [drive.domain.fr] acme: could not determine solvers`

And here is the DNS record:
`*.drive IN A x.x.x.x`

From what I've read, it seems to be an issues with the default resolver used by the acme client. Is there any way to change the DNS resolver used by the acme client? 

Thanks in advance! 

edit: 
Added my dns record

**Comment by 0xJacky** [comment]
Can you please try again?

**Reply by AnFerrr** [reply]
Hey, I tried again but still get this: 

`
Getting the certificate, please wait...

Generating private key for registering account

Preparing lego configurations

Creating client facilitates communication with the CA server

Using DNS01 challenge provider

Setting environment variables

Registering user

Obtaining certificate

Cleaning environment variables

fail to obtain: error: one or more domains had a problem: [*.drive.domain.fr] [*.drive.domain.fr] acme: could not determine solvers [drive.domain.fr] [drive.domain.fr] acme: could not determine solvers`

**Comment by 0xJacky** [comment]
oops, lego does not support the application of wildcard domain certificates.

**Reply by AnFerrr** [reply]
I thought so too at first but some sources seem to say otherwise (for instance: https://www.mslinn.com/blog/2023/03/02/lego.html and the doc https://go-acme.github.io/lego/usage/cli/obtain-a-certificate/). And also from some research it seems there is a way to get it to work by specifying a different dns provider.
Just for the sake of trying everything is there a way I could change the lego config or change the arguments of the command call to try that? I've tried to locate config files on the machine where nginx-ui is installed but I wasn't able to find any

**Comment by Hintay** [comment, marked as answer]
Hi, we will add customized dns reslovers for dns challenge in next version.

**Reply by AnFerrr** [reply]
Alright well thanks for the info :D
I'll just get a cert manually in the mean time! ^^
